File: aspnet-core/src/ERP.Healthcare.Application.Contracts/Patients/UpdatePatientDto.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace ERP.Healthcare.Patients
{
    public class UpdatePatientDto
    {
        [Required]
        [StringLength(PatientConsts.MaxNameLength)]
        public string Name { get; set; }

        [Required]
        public DateTime BirthDate { get; set; }

        [Required]
        [StringLength(PatientConsts.MaxShortDescriptionLength)]
        public string ShortDescription { get; set; }
    }
}

File: angular/src/app/patient/models/update-patient-dto.ts

export class UpdatePatientDto  {
  name: string;
  birthDate: string;
  shortDescription: string;
  doctorId: number;

  constructor(initialValues: Partial<UpdatePatientDto> = {}) {
    if (initialValues) {
      for (const key in initialValues) {
        if (initialValues.hasOwnProperty(key)) {
          this[key] = initialValues[key];
        }
      }
    }
  }
}
